### PR TITLE
SW-3909 Fix search and rendering of boolean in planting progress table

### PR DESCRIPTION
**Pull request by karthikbtf**
- bump web-components to get boolean rendering fix
- fix search to show the correct number of columns (this is a one-time setting for the org not for every search data change)
- update selector to filter out sites that have 0 plantings progress

### CODE REVIEW

File: src/redux/features/plantings/plantingsSelectors.ts [hunk 1]
```diff
@@ -89,24 +91,22 @@ export const searchPlantingProgress = createSelector(
...
-            if (matchesQuery && matchesPlantingCompleted) {
-              acc.push({ siteId, siteName, totalPlants, ...progress });
-            }
-          });
-        } else if (!query && plantingCompleted === undefined) {
-          acc.push({ siteId, siteName, totalSeedlingsSent: totalPlants });
-        }
-        return acc;
-      }, [] as any[]);
+    return plantingProgress?.reduce((acc, curr) => {
```
Comment (karthikbtf) [inline src/redux/features/plantings/plantingsSelectors.ts:94]: this was a one line change (removed line 93 in previous code).. code formatter made the rest of the changes

File: src/redux/features/plantings/plantingsSelectors.ts [hunk 2]
```diff
@@ -58,25 +58,27 @@ export const selectPlantingProgress = createSelector(
...
-                zoneName: zone.name,
-                targetPlantingDensity: zone.targetPlantingDensity,
-                totalSeedlingsSent: plantingsBySubzone[sz.id],
-              }))
-          ),
-          totalPlants: totalPlantsBySite[ps.id],
-        };
-      });
+      return plantingSites
+        ?.filter((ps) => totalPlantsBySite[ps.id])
```
Comment (karthikbtf) [inline src/redux/features/plantings/plantingsSelectors.ts:62]: this is a one line change (added filter) rest of the change is code indent by formatter

File: src/components/NurseryWithdrawals/PlantingProgressList.tsx [hunk 3]
```diff
@@ -73,19 +74,24 @@ export type PlantingProgressListProps = {
...
-  if (!data) {
+  if (!data || hasZones === undefined) {
     return <CircularProgress sx={{ margin: 'auto' }} />;
   }
 
   return (
     <Box>
       <Table
-        id='plantings-progress-table'
+        id={`plantings-progress-table-${hasZones}`}
```
Comment (constanzauanini) [inline src/components/NurseryWithdrawals/PlantingProgressList.tsx:94]: Would this id be planting-progress-table-true and planting-progress-table-false? Why do we need this id?

**Review by constanzauanini — APPROVED** [review]
Working perfectly! Just one question

File: src/components/NurseryWithdrawals/PlantingProgressList.tsx (hunk 3 above)
Comment (karthikbtf) [reply src/components/NurseryWithdrawals/PlantingProgressList.tsx:94]: We use this id to store user preferences for the table column ordering. I can rename it to with/without-zones

Comment (karthikbtf) [reply src/components/NurseryWithdrawals/PlantingProgressList.tsx:94]: See https://github.com/terraware/terraware-web/blob/main/src/components/common/table/index.tsx#L54

Comment (constanzauanini) [reply src/components/NurseryWithdrawals/PlantingProgressList.tsx]: Thank you!!

**Pull request merged**